### PR TITLE
[FW-1053] Add integration tests for XPM2 Draw API

### PR DESCRIPTION
**Pull request by iMalkav**
# What's new

- Added integration coverage for the XPM2 bitmap element introduced in #948.
- Covered rendering on both front and back displays.
- Covered RGB colors, luma conversion, transparency, and opacity.
- Covered malformed XPM2 data and API limits for dimensions, color count, and characters per pixel.
- Covered boundary values, mixed-element requests, replacement, clearing, and redrawing.

# Verification 

- Ran the XPM2 integration test suite against a physical device:

  `poetry run pytest integration/frontend/display/test_api_display_xpm.py -v`

- Result: `24 passed`.

# Checklist (For Reviewer)

- [ ] PR has description of feature/bug or link to Confluence/Jira task
- [ ] Description contains actions to verify feature/bugfix
- [ ] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

**Review by hedger — DISMISSED** [review]

File: tests/integration/frontend/display/test_api_display_xpm.py [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+        x, y = 3, 4
+        blue_data = "! XPM2\n1 1 1 1\nB c #0000FF\nB"
+        elements = [
+            _rectangle("background", "#FF0000FF", x=x, y=y),
+            _xpm(blue_data, x=x, y=y, opacity=0),
+        ]
+
+        response = _draw(assets_api, elements)
+        assets_api.assert_status(response, 200)
+        time.sleep(_RENDER_SETTLE)
```
Comment (ssecsd) [inline tests/integration/frontend/display/test_api_display_xpm.py]: Could we use _wait_for_front_frame instead?

File: tests/integration/frontend/display/test_api_display_xpm.py [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+            response = _draw(
+                assets_api, [_xpm(red_data, x=x, y=y, opacity=50)]
+            )
+            assets_api.assert_status(response, 200)
+            time.sleep(_RENDER_SETTLE)
+            half = _front_pixel(
+                streaming_api.get_screen_bytes(display=0), x, y
+            )
+
+        with allure.step("Verify opacity reduced only the red channel"):
```
Comment (ssecsd) [inline tests/integration/frontend/display/test_api_display_xpm.py:480]: Let's fill bg black before drawing as we expect it being black at the start.

File: tests/integration/frontend/display/test_api_display_xpm.py [hunk 3]
```diff
@@ -0,0 +1,491 @@
...
+    return base
+
+
+def _front_pixel(frame: bytes, x: int, y: int) -> tuple[int, int, int]:
+    """Return one RGB pixel from the front display's raw BGR byte layout."""
+    offset = (y * FRONT_DISPLAY_WIDTH + x) * 3
+    return frame[offset + 2], frame[offset + 1], frame[offset]
+
+
+def _back_pixel(frame: bytes, x: int, y: int) -> int:
```
Comment (ssecsd) [inline tests/integration/frontend/display/test_api_display_xpm.py]: Is it the same as in test_api_display_priority.py:142? Worth move to separate drawing helper?
test_api_display_priority also have _solid_rectangle very similar to _rectangle

File: tests/integration/frontend/display/test_api_display_xpm.py [hunk 4]
```diff
@@ -0,0 +1,491 @@
...
+            frame = streaming_api.get_screen_bytes(display=1)
+            actual = (_back_pixel(frame, x, y), _back_pixel(frame, x + 1, y))
+            assert actual == (0x0, 0xF), (
+                f"Unexpected back XPM pixels: {actual!r}"
+            )
+
+    @allure.title("XPM bitmap requires data")
+    @pytest.mark.api
+    @pytest.mark.frontend
+    def test_missing_data(self, assets_api: AssetsAPI, busy_timer_stopped):
```
Comment (ssecsd) [inline tests/integration/frontend/display/test_api_display_xpm.py:174]: Test lack of allure steps

File: tests/integration/frontend/display/test_api_display_xpm.py [hunk 5]
```diff
@@ -0,0 +1,491 @@
...
+            response = _draw(
+                assets_api,
+                [_xpm(self._BACK_DATA, display="back", x=x, y=y)],
+            )
+            assets_api.assert_status(response, 200)
+            time.sleep(_RENDER_SETTLE)
+
+        with allure.step("Verify the nibble-packed luma values"):
+            frame = streaming_api.get_screen_bytes(display=1)
+            actual = (_back_pixel(frame, x, y), _back_pixel(frame, x + 1, y))
```
Comment (ssecsd) [inline tests/integration/frontend/display/test_api_display_xpm.py]: I think, it's worth to attach screen data to allure steps

File: tests/integration/frontend/display/test_api_display_xpm.py [hunk 6]
```diff
@@ -0,0 +1,491 @@
...
+    def test_api_limits_rejected(
+        self,
+        assets_api: AssetsAPI,
+        busy_timer_stopped,
+        case: str,
+        data: str,
+        display: str,
+    ):
+        response = _draw(assets_api, [_xpm(data, display=display)])
+        assets_api.assert_status(response, 400)
```
Comment (ssecsd) [inline tests/integration/frontend/display/test_api_display_xpm.py]: Lets send valid data that check only one decline at the time. wrong header + valid data and vice versa.

**Review by ssecsd — CHANGES_REQUESTED** [review]

File: tests/integration/frontend/display/test_api_display_xpm.py (hunk 1 above)
Comment (iMalkav) [reply tests/integration/frontend/display/test_api_display_xpm.py]: Add `display_helpers.py`

**Review by ssecsd — APPROVED** [review]
